Lib/test/test_fcntl.py

```diff
@@ -138,6 +136,39 @@ def test_flock(self):
         self.assertRaises(ValueError, fcntl.flock, -1, fcntl.LOCK_SH)
         self.assertRaises(TypeError, fcntl.flock, 'spam', fcntl.LOCK_SH)
 
+    def test_lockf(self):
+
+        self.f = open(TESTFN, 'wb+')
+
+        self.assertRaises(TypeError, fcntl.lockf, self.f, "foo")
+        self.assertRaises(TypeError, fcntl.lockf, self.f, fcntl.LOCK_UN, "foo")
+        self.assertRaises(ValueError, fcntl.lockf, self.f, -256)
+        self.assertRaises(ValueError, fcntl.lockf, self.f, 256)
+
+        fcntl.lockf(self.f, fcntl.LOCK_EX | fcntl.LOCK_NB)
+
+        pid = os.fork()
+        if pid == 0:
+            rval = 2
+            try:
+                fcntl.lockf(open(self.f.name, self.f.mode), fcntl.LOCK_EX | fcntl.LOCK_NB)
+            except IOError as e:
+                if e.errno not in (errno.EACCES, errno.EAGAIN):
+                    raise
+                rval = 0
+            else:
+                rval = 1
+            finally:
+                os._exit(rval)
+
+        assert pid > 0
+        (pid, status) = os.waitpid(pid, 0)
+        self.assertEqual(os.WIFEXITED(status), True)
+
+        fcntl.lockf(self.f, fcntl.LOCK_UN)
+
+        self.f.close()
```

Comment:
I feel this is redundant since `tearDown` takes care of closing this. I am not sure some of the other tests close it too. Is it intentional?